web/src/compose_closed_ui.ts

```diff
@@ -33,46 +33,67 @@ function get_stream_recipient_label(stream_id: number, topic: string): Recipient
     return undefined;
 }
 
-type ComposeClosedMessage = {
+export type ReplyRecipientInformation = {
     stream_id?: number | undefined;
-    topic?: string;
+    topic?: string | undefined;
     display_reply_to?: string | undefined;
 };
 
-export function get_recipient_label(message?: ComposeClosedMessage): RecipientLabel | undefined {
-    // TODO: This code path is bit of a type-checking disaster; we mix
-    // actual message objects with fake objects containing just a
-    // couple fields, both those constructed here and potentially
-    // passed in.
+export function get_recipient_label(
+    recipient_information?: ReplyRecipientInformation,
+): RecipientLabel | undefined {
+    if (recipient_information !== undefined) {
+        // If we're in either the Inbox or Recent Conversations view,
```

Comment:
We can do an assertion here to ensure that remains true with future changes.

Reply:
Yeah, I could see that being a useful follow-up. Not going to block merging on it.

Reply:
Opened #34174 to add that assertion.